test/test_broker.hpp

```diff
@@ -839,6 +846,37 @@ class test_broker {
 
         auto& ep = *spep;
 
+        std::vector<bool> new_sub;
```

Comment:
Recommend making a member variable, so that after the broker has been online for a while, it doesn't have to do new allocations every time.

Reply:
It is temporary variable. SUBSCRIBE packet has multiple entries. It is very small number basically. new_sub marks which entry is new one.  I believe that it is better for temporary variable.